File: problems/greedy_algorithms/tandem_bicycle/js/tandem_bicycle.js
export const tandemBicycle = (redShirtSpeeds, blueShirtSpeeds, fastest) => {
  redShirtSpeeds.sort((a, b) => a - b);
  blueShirtSpeeds.sort((a, b) => a - b);

  if (fastest) {
    redShirtSpeeds.reverse();
  }

  let totalSpeed = 0;
  for (let i = 0; i < redShirtSpeeds.length; i++) {
    totalSpeed += Math.max(redShirtSpeeds[i], blueShirtSpeeds[i]);
  }

  return totalSpeed;
};

File: problems/greedy_algorithms/tandem_bicycle/python/tandem_bicycle.py
def tandemBicycle(
    redShirtSpeeds: list[int], blueShirtSpeeds: list[int], fastest: bool
) -> int:
    redShirtSpeeds.sort()
    blueShirtSpeeds.sort()

    if fastest:
        redShirtSpeeds.reverse()

    totalSpeed = 0
    for i in range(len(redShirtSpeeds)):
        totalSpeed += max(redShirtSpeeds[i], blueShirtSpeeds[i])

    return totalSpeed
